simd/keccakf1600/f1600x4.go

```diff
@@ -0,0 +1,30 @@
+// f1600x4 implements a fast fourway interleaved KeccaK-f[1600] permutation on
+// systems that support it.  KeccaK-f[1600] is the permutation underlying
+// KeccaK, SHA3 and SHAKE.
+//
+// This parallel f[1600] is useful in a few niche applications: for instance,
+// when computing many parallel SHAKE-128 hashes with same-length inputs such
+// as is done in Dilithium or XMSS.
+//
+// Currently only amd64 systems with AVX2 are supported.  This package does
+// not provide a generic implementation as it would be signficantly slower
+// than using non-interleaved f[1600].  Check keccakf1600.AvailableX4 to see
+// if the current system is supported.
+package keccakf1600
+
+import (
+	"github.com/cloudflare/circl/internal/shake"
+)
+
+func genericF1600x4(a []uint64) {
```

Comment:
This function should be in the _generic go file.

Reply:
No: need to access it if AVX2 is unavailable, and `_generic` is only included on `!amd64`.